pandas/core/reshape/merge.py

```diff
@@ -1686,6 +1686,9 @@ def _get_join_indexers(self):
 
         def flip(xs):
             """ unlike np.transpose, this returns an array of tuples """
+            xs = [
+                x if not is_extension_array_dtype(x) else x._ndarray_values for x in xs
+            ]
```

Comment:
@mroeschke can you comment on why _ndarray_values was chosen here?  does it matter that its really lossy for e.g. PeriodIndex?

Reply:
I don't recall. I think `_values` or some equivalent would be okay here